### PR TITLE
[xpu] add xpu custom ops support for llama2-7b

### PR DESCRIPTION
**Pull request by NeroLoh**
### PR types
Others

### PR changes
Others

### Description
add xpu custom ops support for llama2-7b

### CODE REVIEW

File: csrc/xpu/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# ernie-bot-custom-ops
+ernie bot 昆仑自定义算子库。
+
+## 快速开始
+# 构建 XDNN plugin 和 Paddle 自定义算子库
+```
+$ cd src
+# 修改 ./cmake_build.sh 脚本设置 XDNN, XRE and XTDK 的路径后一键执行。
```
Comment (zhink) [inline csrc/xpu/README.md]: XDNN, XRE and XTDK是否能提供获取方法

File: csrc/xpu/src/cmake_build.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
```
Comment (zhink) [inline csrc/xpu/src/cmake_build.sh]: 所有的2023->2024

File: csrc/xpu/src/cmake_build.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+set -e
+
+# export XDNN_PATH=Paddle/build/third_party/xpu/src/extern_xpu/xdnn-ubuntu_x86_64/ # <path_to_xdnn>
+# export XRE_PATH=Paddle/build/third_party/xpu/src/extern_xpu/xre-ubuntu_x86_64/  # <path_to_xre>
+# export CLANG_PATH=xtdk-ubuntu_1604_x86_64 # <path_to_xtdk>
+# export HOST_SYSROOT=/opt/compiler/gcc-8.2/bin/gcc # <path_to_gcc>
+
+cd plugin
+./build.sh
```
Comment (zhink) [inline csrc/xpu/src/cmake_build.sh]: plugin无build.sh

Comment (NeroLoh) [reply csrc/xpu/src/cmake_build.sh]: done

File: csrc/xpu/src/cmake_build.sh (hunk 2 above)
Comment (NeroLoh) [reply csrc/xpu/src/cmake_build.sh]: done

File: csrc/xpu/README.md (hunk 1 above)
Comment (NeroLoh) [reply csrc/xpu/README.md]: done

File: csrc/xpu/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+## 如何贡献
+```
+$ pip install pre-commit==2.17.0
+$ pre-commit install
+```
+
+## 讨论
+如果遇到问题，可以联系 luowei14@baidu.com, zhupengyang@baidu.com, shentanyue01@baidu.com, jiangfan06@baidu.com 或 hongming@baidu.com 解决。
```
Comment (vivienfanghuagood) [inline csrc/xpu/README.md]: remove this

File: csrc/xpu/src/get_output.cc
Comment (vivienfanghuagood) [inline csrc/xpu/src/get_output.cc:1]: maybe can reuse the file from gpu directory?

Comment (NeroLoh) [reply csrc/xpu/src/get_output.cc:1]: done

File: csrc/xpu/README.md (hunk 4 above)
Comment (NeroLoh) [reply csrc/xpu/README.md]: done

File: csrc/xpu/test/python/test_get_padding_offset_v2.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import paddle
+from paddlenlp_ops import get_padding_offset_v2
```
Comment (ZHUI) [inline csrc/xpu/test/python/test_get_padding_offset_v2.py:18]: 最好可以写成 unittest 的格式，方便运行。
```python
class Test(unittest.TestCase):
    pass
```

**Review by ZHUI — DISMISSED** [review]

File: csrc/xpu/test/python/test_get_padding_offset_v2.py (hunk 5 above)
Comment (NeroLoh) [reply csrc/xpu/test/python/test_get_padding_offset_v2.py:18]: done

**Review by ZHUI — APPROVED** [review]
LGTM

**Pull request merged**